### PR TITLE
Added Script to Verify Connection Namespaces on DBaaSPolicy

### PR DESCRIPTION
**Pull request by mrrajan**
Signed-off-by: Rajan Ravi <rravi@redhat.com>

This PR includes,

- Test scripts to verify DBaaSPolicy with connection namespace configuration and their exposure based on user role.
- DBaaS Suite file refactoring based on functionality

### CODE REVIEW

File: resources/keywords/dbaas_policy.resource [hunk 1]
```diff
@@ -100,16 +103,91 @@ Click If Element Present
...
+    Set Test Variable   ${operatorNamespace}    ${namespace}
     User Navigates To DBaaSPolicy Screen    ${namespace}
-    Click If Element Present    ${lblOperand}
-    Wait Until Page Contains Element    ${btnCreateDBaaSPolicy}
-    Click Element    ${btnCreateDBaaSPolicy}
-    Create Custom DBaaSPolicy as ${DBaaSPersona}
+    User Creates DBaaSPolicy For ${DBaaSPersona}
 
 User Creates DBaaSPolicy For ${namespace} From Invalid Namespace Alert on Dev View As ${DBaaSPersona}
+    Set Test Variable   ${operatorNamespace}    ${namespace}
```
Comment (ikanias) [inline resources/keywords/dbaas_policy.resource:111]: What is this ${operatorNamespace}?

File: resources/keywords/dbaas_policy.resource [hunk 2]
```diff
@@ -100,16 +103,91 @@ Click If Element Present
...
     Create Custom DBaaSPolicy as ${DBaaSPersona}
     User Selects Administrator View
 
+User Creates DBaaSPolicy For ${DBaaSPersona}
+    Click If Element Present    ${lblOperand}
+    Wait Until Page Contains Element    ${btnCreateDBaaSPolicy}
+    Click Element    ${btnCreateDBaaSPolicy}
+    Create Custom DBaaSPolicy as ${DBaaSPersona}
+
+Max Quota Exceeded Error Displays
```
Comment (ikanias) [inline resources/keywords/dbaas_policy.resource]: Can we change it to 'Max Quota Exceeded Error Displayed'?

File: resources/keywords/dbaas_policy.resource [hunk 3]
```diff
@@ -100,16 +103,91 @@ Click If Element Present
...
+    IF      ${ns}
+        Click Element    ${btnToggleConnNS}
+        Add Policy Connection Namespaces    ${addNS}
+    END
+    IF    ${label}
+        Click Element    ${btnToggleConnNSSelect}
+        Element Should Be Visible    ${btnToggleMatchExp}
+        Click Element    ${btnToggleMatchExp}
+        Element Should Be Visible    ${btnAddMatchExp}
+        Add Policy Connection Nsselector    ${addLabel}
```
Comment (ikanias) [inline resources/keywords/dbaas_policy.resource]: Change 'Nsselector' to 'NsSelector'

File: resources/keywords/dbaas_policy.resource [hunk 4]
```diff
@@ -100,16 +103,91 @@ Click If Element Present
...
+    END
+
+User Creates DBaaSPolicy With ConnectionNamespace For ${namespace} As ${DBaaSPersona}
```
Comment (ikanias) [inline resources/keywords/dbaas_policy.resource:143]: Should we mention in which view the ConnectionNamespace is created?

File: resources/keywords/dbaas_policy.resource [hunk 5]
```diff
@@ -100,16 +103,91 @@ Click If Element Present
...
+
+Set ConnectionNamespace For DBaaSPolicy
+    Set Test Variable   ${nsAndLabel}   namespace
+    Set Test Variable   @{addNS}    ${connectionNS}
+
+DBaaSInventory Is Available For ${targetns}
+    User Filters Project ${targetns} On Project DropDown And Navigates To Database Access Page
+    Imported Provider Account Available On Database Access Screen
+    User Navigates To Add ${DBProvider} To Topology Screen
+    Select Given Provider Account On Add Topology Screen
```
Comment (ikanias) [inline resources/keywords/dbaas_policy.resource:155]: Why not use ${DBProvider} in this line too instead of 'Given Provider Account'?

File: resources/keywords/dbaas_policy.resource [hunk 6]
```diff
@@ -100,16 +103,91 @@ Click If Element Present
...
+
+User Creates DBaaSPolicy As ${user} With ConnectionNamespace And Imports ${databaseProvider} Provider Account
+    The User Logs In To Openshift As ${user}
+    User Creates DBaaSPolicy With ConnectionNamespace For ${DBaaSTestNS} As ${user}
+    User Is Able To Import MongoDB Provider Account For ${DBaaSTestNS}
```
Comment (ikanias) [inline resources/keywords/dbaas_policy.resource]: Shouldn't we use ${databaseProvider} instead of 'MongoDB Provider Account'?

File: resources/keywords/dbaas_policy.resource [hunk 7]
```diff
@@ -100,16 +103,91 @@ Click If Element Present
...
+
+User Creates DBaaSPolicy As ${user} Without ConnectionNamespace And Imports ${databaseProvider} Provider Account
+    The User Logs In To Openshift As ${user}
+    User Creates DBaaSPolicy For ${DBaaSTestNS} From Admin View As ${user}
+    User Is Able To Import MongoDB Provider Account For ${DBaaSTestNS}
```
Comment (ikanias) [inline resources/keywords/dbaas_policy.resource]: Same here...

File: resources/keywords/dbaas_policy.resource [hunk 8]
```diff
@@ -100,16 +103,91 @@ Click If Element Present
...
+
+DBaaSInventory Is Not Available For ${targetns}
+    User Navigates To Add ${DBProvider} To Topology Screen
+    Wait Until Page Contains Element    ${txtNoDBInstance}    timeout=15s
+
+User Creates ${namespace} And Aligns ${access} Role To The ${user}
+    Set Test Variable    ${access}
```
Comment (ikanias) [inline resources/keywords/dbaas_policy.resource:172]: What does ${access} parameter means here? View/Edit permissions?
And shouldn't we mention that the user giving access roles here is the cluster admin or is this obvious?

File: resources/keywords/dbaas_policy.resource [hunk 9]
```diff
@@ -100,16 +103,91 @@ Click If Element Present
...
+    Execute ${cmd} And Raise Exception For Error
+
+The Namespace ${namespace} Is Not Available On The Project Dropdown
+    User Navigates To Installed Operators Under Operators
+    Wait Until Page Contains Element    ${drpDwnProjectSelect}    timeout=15s
+    Click Element    ${drpDwnProjectSelect}
+    Element Should Be Visible    ${txtBxProjectSelect}
+    Input Text    ${txtBxProjectSelect}    ${namespace}
+    ${btnProjectSelect}=    Replace String    ${btnProjectSelect}    <<project>>    ${namespace}
+    Element Should Not Be Visible    ${btnProjectSelect}
```
Comment (ikanias) [inline resources/keywords/dbaas_policy.resource]: Add new line in the end of this file

File: resources/keywords/suite_and_test_setup_teardown.resource [hunk 10]
```diff
@@ -100,16 +114,18 @@ Setup The Test Case
     ...    AND    Set Global Variable    ${DBaaSPolicyConfigured}    ${True}
 
 Configure DBaaS Policy Namespaces
+    [Arguments]     ${ns}=${DBaaSTestNS}
```
Comment (ikanias) [inline resources/keywords/suite_and_test_setup_teardown.resource:117]: Is ${DBaaSTestNS} = redhat.dbaas.operator?

File: resources/keywords/suite_and_test_setup_teardown.resource [hunk 11]
```diff
@@ -128,4 +144,8 @@ Execute ${cmd} And Raise Exception For Error
     ${stdout} =    Execute Command    ${cmd}    True
     ${stdout_str} =    Convert To String    ${stdout}
     ${stdout_str} =    Convert To Lower Case    ${stdout_str}
-    Should Not Contain    ${stdout_str}     error    msg=Command Executed With Error ${stdout_str}
+    Should Not Contain Any    ${stdout_str}     error   none    msg=Command Executed With Error ${stdout_str}
+
+Remove RoleBinding For The ${user} With ${access} To ${ns}
+    ${cmd} =    Set Variable    oc adm policy remove-role-from-user ${access} ${${user}.USERNAME} -n ${ns}
+    Execute ${cmd} And Raise Exception For Error
```
Comment (ikanias) [inline resources/keywords/suite_and_test_setup_teardown.resource:151]: Add a new line in the end of this file

File: test-variables.yaml.example [hunk 12]
```diff
@@ -25,6 +25,10 @@ OCP_LDAP_DEV:
   AUTH_TYPE: "ldap-provider-qe"
   USERNAME: ldap_user
   PASSWORD: abc123XYZ!
+OCP_DBAAS_USER::
```
Comment (ikanias) [inline test-variables.yaml.example:28]: Why did you add it to the yaml.example file?

File: tests/202_create_dbaas_policy_dev.robot [hunk 13]
```diff
@@ -0,0 +1,62 @@
+*** Settings ***
+Documentation       To verify DBaaSPolicy
```
Comment (ikanias) [inline tests/202_create_dbaas_policy_dev.robot]: Maybe it should say 'To verify DBaaSPolicy in Dev view'?

File: tests/203_dbaas_policy_add_connection_namespace.robot [hunk 14]
```diff
@@ -0,0 +1,68 @@
+*** Settings ***
+Documentation       To verify DBaaSPolicy
```
Comment (ikanias) [inline tests/203_dbaas_policy_add_connection_namespace.robot]: Change to 'Verify Creation of DBaaS Policy Connection Namespace'

File: tests/203_dbaas_policy_add_connection_namespace.robot [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+
+*** Test Cases ***
+Scenario: DBaaSInventories On Source Namespace Is Not Available Without ConnectionNamespace And With Role Permission
+    [Tags]    dbaas-test-34    smoke
+    Given User Creates DBaaSPolicy As OCP_LDAP_PROJECT_ADM Without ConnectionNamespace And Imports MongoDB Provider Account
+    When User Creates ${connectionNs} And Aligns admin Role To The OCP_DBAAS_USER
+    And User Aligns admin Role To ${DBaaSTestNS} For OCP_DBAAS_USER
+    Then DBaaSInventory Is Not Available For ${connectionNs}
+
+Scenario: DBaaSInventories On Source Namespace Is Not Available To ConnectionNamespace And Without RBAC Permission
```
Comment (ikanias) [inline tests/203_dbaas_policy_add_connection_namespace.robot:24]: We should change it to '...Without Any RBAC permissions'?

File: tests/203_dbaas_policy_add_connection_namespace.robot [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+
+Scenario: DBaaSInventories On Source Namespace Available To ConnectionNamespaces Accessed By Same User As Project Admin
+    [Tags]    dbaas-test-36    smoke
+    Given User Creates DBaaSPolicy As OCP_LDAP_PROJECT_ADM With ConnectionNamespace And Imports MongoDB Provider Account
+    When User Creates ${connectionNs} And Aligns admin Role To The OCP_LDAP_PROJECT_ADM
+    Then DBaaSInventory Is Available For ${connectionNS}
+
+Scenario: DBaaSInventories On Source Namespace Available To ConnectionNamespaces Accessed By Same User As Service Admin
+    [Tags]    dbaas-test-37    smoke
+    Given User Creates DBaaSPolicy As OCP_LDAP_PROJECT_ADM With ConnectionNamespace And Imports MongoDB Provider Account
```
Comment (ikanias) [inline tests/203_dbaas_policy_add_connection_namespace.robot:38]: Shouldn't this user be OCP_LDAP_SERVICE_ADMIN like in the scenario title?

File: tests/203_dbaas_policy_add_connection_namespace.robot [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+    Then DBaaSInventory Is Available For ${connectionNS}
+
+Scenario: DBaaSInventories On Source Namespace Available To ConnectionNamespaces Accessed By Another User As Project Admin For Both Namespaces
+    [Tags]    dbaas-test-38    smoke
+    Given User Creates DBaaSPolicy As OCP_LDAP_PROJECT_ADM With ConnectionNamespace And Imports MongoDB Provider Account
+    When User Creates ${connectionNs} And Aligns admin Role To The OCP_DBAAS_USER
+    And User Aligns admin Role To ${DBaaSTestNS} For OCP_DBAAS_USER
+    Then DBaaSInventory Is Available For ${connectionNS}
+
+Scenario: DBaaSInventories Available To ConnectionNamespaces Accessed By Another User Being Service Admin For Source and Project Admin For Target Namespace
```
Comment (ikanias) [inline tests/203_dbaas_policy_add_connection_namespace.robot:49]: Can we change the second line to 'Source Namespace'?

File: tests/203_dbaas_policy_add_connection_namespace.robot [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+    [Tags]    dbaas-test-39    smoke
+    Given User Creates DBaaSPolicy As OCP_LDAP_PROJECT_ADM With ConnectionNamespace And Imports MongoDB Provider Account
+    When User Creates ${connectionNs} And Aligns admin Role To The OCP_DBAAS_USER
+    And User Aligns edit Role To ${DBaaSTestNS} For OCP_DBAAS_USER
+    Then DBaaSInventory Is Available For ${connectionNS}
+
+Scenario: DBaaSInventories Available To ConnectionNamespaces Accessed By Different User Being Project Admin For Source and Service Admin For Target Namespace
```
Comment (ikanias) [inline tests/203_dbaas_policy_add_connection_namespace.robot]: Can we change 'Different User' to be 'Another User' for consistency?
And regarding the policy rules, Shouldn't the second user that gets the permissions to use the ConnectionNS should have an equal or lower role than the user creating the ConnectionNS and giving him the roles?

File: tests/203_dbaas_policy_add_connection_namespace.robot [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+    [Tags]    dbaas-test-40    smoke
+    Given User Creates DBaaSPolicy As OCP_LDAP_PROJECT_ADM With ConnectionNamespace And Imports MongoDB Provider Account
+    When User Creates ${connectionNs} And Aligns edit Role To The OCP_DBAAS_USER
+    And User Aligns admin Role To ${DBaaSTestNS} For OCP_DBAAS_USER
+    Then DBaaSInventory Is Available For ${connectionNS}
+
+Scenario: DBaaSInventories On Source Namespace Available To ConnectionNamespaces Accessed By Different User As Service Admin For Both Namespaces
```
Comment (ikanias) [inline tests/203_dbaas_policy_add_connection_namespace.robot]: Same here...

File: utils/scripts/dbaas_utils.py [hunk 20]
```diff
@@ -372,3 +372,35 @@ def create_provision_instance_yaml(isv, prov_acc_ns, deploy_instance_ns):
         data["spec"]["otherInstanceParams"]["projectName"] = instance
     print(data)
     return yaml.dump(data, sort_keys=False)
+
+
+def add_policy_connection_namespaces(namespace: list):
+    """Adding Connection Namespace on DBaaSPolicy"""
+    sl = BuiltIn().get_library_instance("SeleniumLibrary")
+    btn_add_conn_xpath = "//button[@id='root_spec_connectionNamespaces_add-btn']"
```
Comment (ikanias) [inline utils/scripts/dbaas_utils.py]: I think we should put this in the resources/object_repo/create_dbaaspolicy_obj.resource. I think it does not look good being here and we should also be consistent. WDYT?

File: utils/scripts/dbaas_utils.py [hunk 21]
```diff
@@ -372,3 +372,35 @@ def create_provision_instance_yaml(isv, prov_acc_ns, deploy_instance_ns):
...
+    if not namespace:
+        namespace = "*"
```
Comment (ikanias) [inline utils/scripts/dbaas_utils.py:383]: Can you please explain what '*' means here? Does it mean that this ConnectionNS can be seen by all other namespaces?

File: utils/scripts/dbaas_utils.py [hunk 22]
```diff
@@ -372,3 +372,35 @@ def create_provision_instance_yaml(isv, prov_acc_ns, deploy_instance_ns):
...
+    for i, ns in enumerate(namespace):
+        sl.click_element(btn_add_conn_xpath)
+        txtbx_value_xpath = "//input[@id='root_spec_connectionNamespaces_" + str(i) + "']"
```
Comment (ikanias) [inline utils/scripts/dbaas_utils.py]: Same here, we need to move it to  resources/object_repo/create_dbaaspolicy_obj.resource.

File: utils/scripts/dbaas_utils.py [hunk 23]
```diff
@@ -372,3 +372,35 @@ def create_provision_instance_yaml(isv, prov_acc_ns, deploy_instance_ns):
...
+        sl.input_text(txtbx_value_xpath, ns)
+
+
+def add_policy_connection_nsselector(labelandconds: list):
+    """Adding connection Namespace Selector on DBaaSPolicy"""
+    sl = BuiltIn().get_library_instance("SeleniumLibrary")
+    btn_add_matchexp_xpath = "//button[@id='root_spec_connectionNsSelector_matchExpressions_add-btn']"
```
Comment (ikanias) [inline utils/scripts/dbaas_utils.py]: We should move it to resources/object_repo/create_dbaaspolicy_obj.resource.

File: utils/scripts/dbaas_utils.py [hunk 24]
```diff
@@ -372,3 +372,35 @@ def create_provision_instance_yaml(isv, prov_acc_ns, deploy_instance_ns):
...
+    for i, labelAndCond in enumerate(labelandconds):
+        sl.click_element(btn_add_matchexp_xpath)
+        txtbx_key_xpath = "//input[@id='root_spec_connectionNsSelector_matchExpressions_" + str(i) + "_key']"
```
Comment (ikanias) [inline utils/scripts/dbaas_utils.py]: Same here

File: utils/scripts/dbaas_utils.py [hunk 25]
```diff
@@ -372,3 +372,35 @@ def create_provision_instance_yaml(isv, prov_acc_ns, deploy_instance_ns):
...
+        txtbx_oper_xpath = "//input[@id='root_spec_connectionNsSelector_matchExpressions_" + str(i) + "_operator']"
+        btn_values = "//button[@id='root_spec_connectionNsSelector_matchExpressions_" + str(i) + "_values_accordion-toggle']"
+        for cond in labelAndCond.split(","):
+            sl.input_text(txtbx_key_xpath, cond[0])
+            sl.input_text(txtbx_oper_xpath, cond[1])
```
Comment (ikanias) [inline utils/scripts/dbaas_utils.py:406]: Here cond[0] is the value left of the ',' in the split, and cond[1] is the value in the right of ',' right?

**Review by ikanias — CHANGES_REQUESTED** [review]
Had some questions and comments. Please LMKWYT.

File: resources/keywords/dbaas_policy.resource (hunk 1 above)
Comment (mrrajan) [reply resources/keywords/dbaas_policy.resource:111]: ${operatorNamespace} Already in use and I am reusing this variable here

File: resources/keywords/dbaas_policy.resource (hunk 2 above)
Comment (mrrajan) [reply resources/keywords/dbaas_policy.resource]: @ikanias As per Gherkin good practice, we try to maintain present tense wherever possible. IMO, I hope this is fine.

File: resources/keywords/dbaas_policy.resource (hunk 3 above)
Comment (mrrajan) [reply resources/keywords/dbaas_policy.resource]: Method "add_policy_connection_nsselector" is a low level keyword and while calling it as middle level we should use Title case. This change was introduced by Robotidy, even If we change it now the same change would be introduced while running robotidy.

File: resources/keywords/dbaas_policy.resource (hunk 4 above)
Comment (mrrajan) [reply resources/keywords/dbaas_policy.resource:143]: This action performed on DBaaSPolicy screen and irrespective of Openshift view, this would be same for all the flows. So keeping it general, I hope we should leave it as it is.

File: resources/keywords/dbaas_policy.resource (hunk 5 above)
Comment (mrrajan) [reply resources/keywords/dbaas_policy.resource:155]: Existing keyword "Select Given Provider Account On Add Topology Screen" uses variable assignment internally. Please go through the definition and let me know if still this needs to be addressed.

File: resources/keywords/dbaas_policy.resource (hunk 8 above)
Comment (mrrajan) [reply resources/keywords/dbaas_policy.resource:172]: Yes, this is for the RoleBinding for the user and yes only cluster admin can do that which is part of line 174

File: resources/keywords/dbaas_policy.resource (hunk 9 above)
Comment (mrrajan) [reply resources/keywords/dbaas_policy.resource]: I am surprised as the file contains the new line at the end of this file and it is not showing up on github. Even I dont see a warning message for this. Something to learn, but I dont think this would be blocker for this PR. Please let me know.

File: resources/keywords/suite_and_test_setup_teardown.resource (hunk 10 above)
Comment (mrrajan) [reply resources/keywords/suite_and_test_setup_teardown.resource:117]: No, The assignment of variable is done on login.resource file.

File: resources/keywords/suite_and_test_setup_teardown.resource (hunk 11 above)
Comment (mrrajan) [reply resources/keywords/suite_and_test_setup_teardown.resource:151]: Same here

File: test-variables.yaml.example (hunk 12 above)
Comment (mrrajan) [reply test-variables.yaml.example:28]: It should added as this would be reference template for the user. Please let me know if this needs to be addressed, but IMO this should be fine.

File: tests/203_dbaas_policy_add_connection_namespace.robot (hunk 15 above)
Comment (mrrajan) [reply tests/203_dbaas_policy_add_connection_namespace.robot:24]: IMHO, this should be fine as we already have more than the limit.

File: tests/203_dbaas_policy_add_connection_namespace.robot (hunk 16 above)
Comment (mrrajan) [reply tests/203_dbaas_policy_add_connection_namespace.robot:38]: The purpose of this scenario is to create a DBaaSPolicy to expose inventories to another namespace and accessing the another namespace with Service admin access with the same user. 
So on line 39, OCP_LDAP_PROJECT_ADM is given with "edit" role (service admin access) for the namespace mentioned on "ConnectionNamespace" field which fulfills the scenario.
Please ignore the user name as the RBAC configured for namespace.

File: tests/203_dbaas_policy_add_connection_namespace.robot (hunk 17 above)
Comment (mrrajan) [reply tests/203_dbaas_policy_add_connection_namespace.robot:49]: We could, but again we have same constrain on length of the sentence as this already crossed the limit. So IMHO, this should be fine.

File: tests/203_dbaas_policy_add_connection_namespace.robot (hunk 18 above)
Comment (mrrajan) [reply tests/203_dbaas_policy_add_connection_namespace.robot]: - Updated to "Another User"
- For this question, Not necessary. The roles could be only assigned by Cluster admin. And we can have more than one admin for a namespace. By default the one who creates the ns becomes the admin for that ns, which can also be reverted by cluster admin. I hope this would clarify, Please let me know in case of any concerns.

File: utils/scripts/dbaas_utils.py (hunk 21 above)
Comment (mrrajan) [reply utils/scripts/dbaas_utils.py:383]: Yes, only if it is not specified and this method is being invoked.

File: utils/scripts/dbaas_utils.py (hunk 20 above)
Comment (mrrajan) [reply utils/scripts/dbaas_utils.py]: The risk of this keyword is not limited to this object but also other dynamic ones on the other parts. I will include this changes, but it still has the risk and it also introduces additional layers to pull the objects from the resource file.

File: utils/scripts/dbaas_utils.py (hunk 25 above)
Comment (mrrajan) [reply utils/scripts/dbaas_utils.py:406]: Yes

**Comment by mrrajan** [comment]
Thanks for the feedback @ikanias, I made the changes and resolved some of the questions. PTAL and let me know.

But in the future can you please consolidate the same kind of review comments if possible.

File: resources/keywords/dbaas_policy.resource (hunk 3 above)
Comment (ikanias) [reply resources/keywords/dbaas_policy.resource]: OK got it.

File: resources/keywords/dbaas_policy.resource (hunk 2 above)
Comment (ikanias) [reply resources/keywords/dbaas_policy.resource]: What I meant is actually 'Error is being displayed'/ 'Error is displayed' in present tense as passive. What is written here is that the error 'displays' something (active) which is wrong :-)

File: resources/keywords/dbaas_policy.resource (hunk 4 above)
Comment (ikanias) [reply resources/keywords/dbaas_policy.resource:143]: But isn't consistent with the previous keywords.
Shouldn't it be consistent? Or maybe you mean that in this specific keyword only the view does not matter?

File: resources/keywords/dbaas_policy.resource (hunk 5 above)
Comment (ikanias) [reply resources/keywords/dbaas_policy.resource:155]: OK got it.

File: resources/keywords/dbaas_policy.resource (hunk 8 above)
Comment (ikanias) [reply resources/keywords/dbaas_policy.resource:172]: ok got it

File: resources/keywords/dbaas_policy.resource (hunk 4 above)
Comment (mrrajan) [reply resources/keywords/dbaas_policy.resource:143]: Yes @ikanias For this keyword view does not matter.

File: tests/203_dbaas_policy_add_connection_namespace.robot (hunk 16 above)
Comment (ikanias) [reply tests/203_dbaas_policy_add_connection_namespace.robot:38]: ok got it

**Review by ikanias — COMMENTED** [review]
Some more clarifications

File: resources/keywords/dbaas_policy.resource (hunk 9 above)
Comment (ikanias) [reply resources/keywords/dbaas_policy.resource]: Maybe it relates to robotidy?

File: test-variables.yaml.example (hunk 12 above)
Comment (ikanias) [reply test-variables.yaml.example:28]: OK got it

File: resources/keywords/dbaas_policy.resource (hunk 9 above)
Comment (mrrajan) [reply resources/keywords/dbaas_policy.resource]: I dont think so

File: tests/203_dbaas_policy_add_connection_namespace.robot (hunk 18 above)
Comment (ikanias) [reply tests/203_dbaas_policy_add_connection_namespace.robot]: OK got it

File: utils/scripts/dbaas_utils.py (hunk 20 above)
Comment (ikanias) [reply utils/scripts/dbaas_utils.py]: I do not understand what you mean. Are you saying that because this object is defined in the low layer, putting it in resources/object_repo/create_dbaaspolicy_obj.resource could cause issues?

**Comment by ikanias** [comment]
Yes @mrrajan will try to consolidate similar comments :-)

File: resources/keywords/dbaas_policy.resource (hunk 9 above)
Comment (ikanias) [reply resources/keywords/dbaas_policy.resource]: OK strange...

File: utils/scripts/dbaas_utils.py (hunk 20 above)
Comment (mrrajan) [reply utils/scripts/dbaas_utils.py]: It wont cause issues, the risk of maintaining the objects due to changes still lies in low level  as we are dynamically updating the objects with index value.

File: resources/keywords/dbaas_policy.resource (hunk 2 above)
Comment (mrrajan) [reply resources/keywords/dbaas_policy.resource]: Updated phrasing

File: resources/keywords/dbaas_policy.resource (hunk 4 above)
Comment (ikanias) [reply resources/keywords/dbaas_policy.resource:143]: OK got it

**Review by ikanias — APPROVED** [review]
